platform/mds/mds-web/src/main/java/org/motechproject/mds/web/controller/EntityController.java

```diff
@@ -70,7 +70,7 @@
                     byModule.put(entity.getModule(), new ArrayList<EntityDto>());
                 }
 
-                if (!entity.isAbstractClass()) {
+                if (!entity.isAbstractClass() || entity.isDiscriminated()) {
```

Comment:
Re: "Why are we displaying these"

See https://applab.atlassian.net/browse/MOTECH-1649

    Widget appearance:
        - A non-editable label "subclass".
        - A mouse dropdown selector.
        - A choice for every y in Y, plus a choice for undefined.

    When creating an the, a widget allows the user to:

        1) See when y is has not been chosen.
        2) See the name of the subclass when y has been chosen.
        3) Choose y.

But the code fragment indicated is not any of this behavior.  The code fragment's behavior is:

    When the user updates an instance using entityId=x, we edit the instance the same as we would if we updated the instance with the entityId of its derived class.

Added to MOTECH-1649 for clarification.  Will also add a code comment.